File: app/src/main/java/eu/sesma/dagger/di/DetailActivityComponent.java
package eu.sesma.dagger.di;

import eu.sesma.dagger.platform.ActivityModule;
import eu.sesma.dagger.platform.DetailActivityModule;
import eu.sesma.dagger.ui.detail.DetailActivity;

import dagger.Subcomponent;
import dagger.android.AndroidInjector;

@PerActivity
@Subcomponent(modules = { ActivityModule.class, DetailActivityModule.class})
public interface DetailActivityComponent extends AndroidInjector<DetailActivity> {

    @Subcomponent.Builder
    abstract class Builder extends AndroidInjector.Builder<DetailActivity> {
    }
}

File: module_detail/src/main/java/eu/sesma/dagger/detail/di/DetailActivityComponent.kt
package eu.sesma.dagger.detail.di

import android.support.v7.app.AppCompatActivity
import dagger.Component
import eu.sesma.dagger.core.di.ActivityScoped
import eu.sesma.dagger.core.di.CoreComponent
import eu.sesma.dagger.detail.DetailActivity


@ActivityScoped
@Component(
        dependencies = [CoreComponent::class],
        modules = [DetailActivityModule::class]
)
interface DetailActivityComponent {

    fun inject(activity: DetailActivity)

    //Exposed to sub-graphs.
    fun activity(): AppCompatActivity
}